src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -181,11 +181,17 @@ private FunctionScoreQuery getFunctionScoreQuery(
       com.yelp.nrtsearch.server.grpc.FunctionScoreQuery functionScoreQuery, IndexState state) {
     ScoreScript.Factory scriptFactory =
         ScriptService.getInstance().compile(functionScoreQuery.getScript(), ScoreScript.CONTEXT);
+
     Map<String, Object> params =
         ScriptParamsUtils.decodeParams(functionScoreQuery.getScript().getParamsMap());
-    return new FunctionScoreQuery(
-        getQuery(functionScoreQuery.getQuery(), state),
-        scriptFactory.newFactory(params, state.docLookup));
+    com.yelp.nrtsearch.server.grpc.Query q = functionScoreQuery.getQuery();
+    Query query;
+    if (!q.toString().isEmpty()) {
+      query = getQuery(q, state);
+    } else {
+      query = new MatchAllDocsQuery();
+    }
```

Comment:
We have some similar logic done for the root query https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java#L254 
@sarthakn7 do you think maybe this should be the default behavior of the QueryNodeMapper whenever there is an UNSET query node?

Reply:
+1 Yeah we can make it the default behavior. That way this would be handled consistently for any other query that nests a query.